packages/frontend/src/components/DashboardTiles/DashboardChartTile.tsx

```diff
@@ -300,7 +302,7 @@ const DashboardChartTileMain: FC<DashboardChartTileMainProps> = (props) => {
         DashboardFilterRule[]
     >([]);
     const [viewUnderlyingDataOptions, setViewUnderlyingDataOptions] = useState<{
-        item: Field | TableCalculation | undefined;
+        item: Field | Metric | AdditionalMetric | TableCalculation | undefined;
```

Comment:
Changing this `item` interface in many places makes me nervous about the impact of this PR. Why do we need to add this? What's different about pie charts that it cannot use the existing types?

Reply:
we are viewing underlying data of a metric and metric can be AdditionalMetric.

They have almost the same type interface but the key difference is that Metric extends Field but AdditionalMetric does not.

most likely somewhere we have a missing type of AdditionalMetric but in this case here I believe changing types makes sense

Reply:
The way we've previously done this is with `convertAdditionalMetric` 

for example in the bignumberconfig: https://github.com/lightdash/lightdash/blob/main/packages/frontend/src/hooks/useBigNumberConfig.ts#L95-L95

Might not be the best way but it's consistent, and if it's consistent it's easier to refactor later

Reply:
updated!